File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        Map = {")": "(", "]": "[", "}": "{"}
        stack = []
        for c in s:
            if c not in Map:
                stack.append(c)
                continue
            elif (len(stack) != 0 and Map[c] == stack.pop()):
                continue
            else:
                return False
        return not stack


        

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.java
class Solution {
    public int evalRPN(String[] tokens) {
        
        ArrayDeque<Integer> stack = new ArrayDeque<>();

        int i = 0;
        while (i < tokens.length) {
            if ("+-*/".contains(tokens[i])) {
                Integer left = stack.pop();
                Integer right = stack.pop();
                Integer output = 0;
                switch(tokens[i]) {
                    case "+":
                        output = left + right;
                        break;
                    case "-":
                        output = right - left;
                        break;
                    case "*":
                        output = left * right;
                        break;
                    case "/":
                        output = right / left;
                        break;
                }
                System.out.println("left:" + left);
                System.out.println("right:" + right);
                System.out.println("output:" + output);
                stack.push(output);

            } else {
                System.out.println(Integer.valueOf(tokens[i]));
                stack.push(Integer.valueOf(tokens[i]));
            }
            i++;
        }
        return stack.pop();
    }
}
